Add support for ZHA door locks

presslab-us opened the pull request:
## Description:

Adds support for ZHA door locks.  Supports lock and unlock.

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#9522

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [ ] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

pvizeli committed:
Fix Hassio-version for Azure Pipelines (#23895)

balloob committed:
Update Honeywell warning (#23913)

elupus committed:
Switch media player to SWITCH type (#23914)

MEDIA device type is being rejected by google now.

karlkar committed:
Fix problem with cameras that don't support time (#23924)

Some onvif cameras don't support Date management. In that case None is returned and script crashes when trying to obtain date

balloob committed:
Bumped version to 0.93.1

balloob committed:
Merge pull request #23932 from home-assistant/rc

0.93.1

pvizeli committed:
Fix auto version update Hass.io (#23935)

presslab-us committed:
Add support for DoorLock cluster

presslab-us committed:
Add support for DoorLock cluster

presslab-us committed:
Add test for zha lock

presslab-us committed:
Add test for zha lock

ludeeus committed:
Bump pytraccar (#23939)

presslab-us committed:
Change lock_state report to REPORT_CONFIG_IMMEDIATE

presslab-us committed:
Change lock_state report to REPORT_CONFIG_IMMEDIATE

aerialls committed:
Setup integration dependencies before loading it (#23957)

Danielhiversen committed:
upgrade broadlink library (#23966)

cyrosy committed:
Make Discord payload data key not required (#23964)

ludeeus committed:
Fixes issue with multiple alerts (#23945)

* Fixes issue with multiple alerts

* Adds missing new line

* Remove whitespace

balloob committed:
Bumped version to 0.93.2

balloob committed:
Merge pull request #24047 from home-assistant/rc

0.93.2

presslab-us committed:
Update channel command wrapper to return the entire result
This allows for return values other than result[1]

presslab-us committed:
Update channel command wrapper to return the entire result
This allows for return values other than result[1]

presslab-us committed:
Fix tests

presslab-us committed:
Fix tests

presslab-us committed:
Fix lint

presslab-us committed:
Fix lint

homeassistant commented:
Hi @presslab-us, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#24126).

Once you do that we will be able to review and accept this pull request.

Thanks!

balloob committed:
Use importlib metadata to check installed packages (#24114)

* Use importlib metadata

* Fix script

* Remove unused import

* Update requirements"

amelchio committed:
Avoid useless Sonos state updates (#24135)

balloob committed:
Add SSDP integration (#24090)

* Add SSDP integration

* Fix tests

* Sort all the things

* Add netdisco to test requirements

bramkragten committed:
Lovelace: Fire event on save (#24104)

* Lovelace: Fire event on save

* Add event to whitelist

jjlawren committed:
Use central polling to update entities (#24059)

* Use central polling to update entities

* Fix for line length

* Remove unnecessary import

* Use interval

* Trigger entity refreshes after commands

* Lint

Kane610 committed:
Library refactorization of deCONZ (#23725)

* Improved sensors

* Lib update signalling

* Replace reason with changed

* Move imports to top of file

* Add support for secondary temperature reported by some Xiaomi devices

* Bump dependency to v59

aerialls committed:
Retrieve wire and wireless devices with the SRM device tracker (#24117)

maheus committed:
Add station name for creating the unique_id in netatmo platform (#24141)

jesserizzo committed:
bump dependency envoy_reader to 0.4 (#24145)

* bump envoy_reader version to 0.4

* bump dependency envoy_reader to 0.4

TheLastProject committed:
Don't crash on first EAN without installations (#24137)

* Don't crash on first EAN without installations

* Remove duplicated values

* Switch from Exception to persistent notification

* Make pylint happy

Commit:
pubnubsub-handler to 1.0.5 (#24154)

Commit:
Pubnub to 1.0.6 (#24159)

Adminiuga committed:
Debug log when polling ZHA light. (#24167)

chmielowiec committed:
Upgrade huawei-lte-api to 1.2.0 (#24165)

robbiet480 committed:
Use device name for device_tracker entry (#24155)

OttoWinter committed:
Use global imports for ESPHome (#24158)

* Use global import for ESPHome

* Add aioesphomeapi to test requirements

aioesphomeapi is also shipped as a pure-python wheel, so this should not impact test install time

MTrab committed:
Add Repetier-Server Component (#21658)

* Bumped to version 2.0

* Updated requirements

* Updated requirements and coveragerc

* Removed long lines, changes to coveragerc and requirements

* Fixed under-indented lines

* Fixed invalid syntax

* Updated .coveragerc to include repetier/__init__.py and sensor.py

* Module update

* Rebased to latest dev

* Blank lines fix

* Add missing manifest.json

* Update requirements

* Bumped component to new API module style

* Removed whitespaces and line feeds

* Added missing newline

* Added missing heated chamber sensor

* Fixed wrong indentation

* Various fixes

* Various build fixes

* Clean up

* Load platform only once

* Sort imports

* Add printer api

* Clean up

* Build out sensor classes

* Clarify temperature sensor variable names

* Move constants

* Clean up name

* Run script/gen_requirements_all.py

* Working code, missing auto add of new sensors

* Updated code to return proper device_class and timestamp

* Removed unnessecary code

* Renamed elapsed and remaining sensors

* Dynamically adding sensors as they become available

* Rebased .coveragerc due to conflicts

* Code changes and cleanup

* Removed whitespace and code simplification

balloob committed:
Cloud: Websocket API to manage Google assistant entity config (#24153)

* Extract exposed devices function

* Add might_2fa info to trait

* Do not filter with should_expose in Google helper func

* Cloud: allow setting if Google entity is exposed

* Allow disabling 2FA via config

* Cloud: allow disabling 2FA

* Lint

* More changes

* Fix typing

robbiet480 committed:
Fix calling notify.notify with mobile_app targets in play. Fixes #24064 (#24156)

balloob committed:
Updated frontend to 20190529.0

balloob committed:
update translations

amelchio committed:
Remove unused Sonos turn on/off methods (#24174)

balloob committed:
Reinstate passing loop to DSMR (#24127)

* Reinstate passing loop

* Also pass loop into other part

Kane610 committed:
Trådfri component to use new zeroconf discovery (#24041)

* Move tradfri component to use new zeroconf discovery

* Will this work?

* Remove prints

* Correct order in generated zeroconf

* Update test_init.py

* Update test_init.py

* Update test_init.py

* Update test_init.py

Kane610 committed:
Move Homekit controller component to user zeroconf discovery (#24042)

Kane610 committed:
Revert Zeroconf back to previously used library (#24139)

* Revert back to previously used library

* Fix test

* Remove unused import

* Fix import

* Update __init__.py

* Update __init__.py

* Fix test after rebase

balloob committed:
Deprecate Python 3.5.3 (#24177)

Kane610 committed:
Keep integrations in discovery (#24179)

* Keep integrations that have been migrated to new discovery methods to avoid breaking changes

* Additional migrated services

amelchio committed:
Avoid slow updates with unavailable Sonos devices (#24180)

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

pvizeli committed:
Support Hass.io wheels / docker env (#24175)

* Support Hass.io wheels / docker env

* address comments

* fix lint

balloob committed:
Remove discovery from initial config (#24183)

Jc2k committed:
Fix duplicated discovered homekit devices (#24178)

alengwenus committed:
Add service calls for LCN component (#24105)

amelchio committed:
Update pysonos to 0.0.14 (#24185)

Danielhiversen committed:
upgrade broadlink library, Use cryptography instead of pycryptodome (#24186)

Danielhiversen committed:
danielhiversen as codeowner for yr (#24189)

balloob committed:
No longer allow invalid slugs or extra keys (#24176)

* No longer allow slugs

* Lint

* Remove HASchema

* Lint

* Fix tests

balloob committed:
Dynamic panels (#24184)

* Don't require all panel urls to be registered

* Allow removing panels, fire event when panels updated

pvizeli committed:
Update azure-pipelines.yml for check version (#24194)

balloob committed:
Updated frontend to 20190530.0

Jc2k committed:
homekit_controller no longer logs with transient network errors causing crypto failures as it will auto recover (#24193)

Danielhiversen committed:
Update Tibber library (#24192)

frenck committed:
:pencil2: Corrects incorrect command in hassfest (#24188)

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

OttoWinter committed:
Fix ESPHome discovered when already exists (#24187)

* Fix ESPHome discovered when already exists

* Update .coveragerc

balloob committed:
Update hass-nabucasa (#24197)

balloob committed:
Allow discovery flows to be discovered via zeroconf/ssdp (#24199)

balloob committed:
Improve error handling (#24204)

vangorra committed:
Fixing tplink issues with offline devices during setup (#23668)

* Fixing tplink issues with offline devices during setup.

* Fixing circleci errors.

* Adding code to defer the creation of entities that are not online.

* Addressing code review changes and cleaning up a little.

* Fixing tests and static analysis.

* Adding test to satisfy coverage requirements.

* Resolving merge conflicts.

* Fixing issue where lights don't appear in the integration page.

* Using pyHS100 properties for most sysinfo.
Addressing some PR feedback.

* Addressing some PR feedback.

* Better testing async_add_entities_retry
Testing for static dimmers.
Making greater use of conf constants.

* Fixing all static analysis issues.

* Adding non-blocking call for getting discovering devices.

* Address PR feedback

therve committed:
Bump oauthlib version (#24111)

* Bump oauthlib version

* Bump fitbit instead

* Update requirements

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

OttoWinter committed:
Fix ESPHome config flow with invalid config entry (#24213)

pvizeli committed:
Don't follow redirect on ingress itself (#24218)

* Don't follow redirect on ingress itself

* Fix comment

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

balloob committed:
Instantiate lock inside event loop (#24203)

balloob committed:
Use resource for index routing. (#24223)

Kane610 committed:
Axis - Handle Vapix error messages (#24215)

balloob committed:
Add manifest support for homekit discovery (#24225)

* Add manifest support for homekit discovery

* Add a space after model check

* Update comment

pnbruckner committed:
Add control of Amcrest indicator light (#23986)

Enable feature by default but allow it to be disabled by "control_light: false" in config.

Get brand from camera instead of assuming Amcrest (since this works with other cameras, too.)

Retrieve RTSP URL in update method instead of in stream_source property and in handle_async_mjpeg_stream method.

Move amcrest imports from methods to global.

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

Adminiuga committed:
ZHA requirements version bump. (#24228)

* ZHA requirements version bump.
* zha-quirks version bump.

Kane610 committed:
Don't allow more than one config flow per discovered Axis device (#24230)

soldag committed:
Bump pychromecast (#24234)

balloob committed:
Log HomeKit model (#24229)

balloob committed:
Add GPSLogger device_info and unique_id (#24231)

balloob committed:
GeoFency unique ID and device info (#24232)

rytilahti committed:
add a deprecation warning for tplink device_tracker (#24236)

* add a deprecation warning for tplink device_tracker

* reword the warning a bit

balloob committed:
Mobile app to use device tracker config entry (#24238)

* Mobile app to use device tracker config entry

* Lint

* Re-use device_info

* Lint

balloob committed:
Update home zone when core config updated (#24237)

* Update home zone when core config updated

* Lint

balloob committed:
Do not use the cache dir for PIP installs (#24233)

danielsjf committed:
Round Awair sensor values (#24093)

* Round sensor values

* Add code owner

* Update code owners

* Fix tests

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

kbickar committed:
Sense update (#24220)

* Changed updates to be done by component and updates dispatched

* syntax updates

* Added code owner

* Removed whitespace

* Updated CODEOWNERS

* Added subscription undoer

Emilv2 committed:
fix gitlab_ci sad icon (#24241)

austinmroczek committed:
Add details to triggered state for total connect alarms (#24106)

* Bump skybellpy to 0.4.0

* Bump skybellpy to 0.4.0 in requirements_all.txt

* Added extra states for STATE_ALARM_TRIGGERED to allow users to know if
it is a burglar or fire or carbon monoxide so automations can take
appropriate actions.  Updated TotalConnect component to handle these new
states.

* Fix const import

* Fix const import

* Fix const imports

* Bump total-connect-client to 0.26.

* Catch details of alarm trigger in state attributes.

Also bumps total_connect_client to 0.27.

* Change state_attributes() to device_state_attributes()

cereal2nd committed:
Bump python-velbus version for velbus component (#24226)

* Bump python-velbus version

* Bump python velbus version

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

balloob committed:
Updated frontend to 20190601.0

frenck committed:
Adds AdGuard Home integration (#24219)

* Adds AdGuard Home integration

* :shirt: Addresses linting warnings

* :ambulance: Fixes typehint in async_setup_entry

* :shirt: Take advantage of Python's coalescing operators

* :shirt: Use adguard instance from outer scope directly in service calls

* :shirt: Use more sensible scan_interval default for sensors

* :shirt: Adds specific files to .coveragerc

* :umbrella: Added tests and small changes to improve coverage

* :hammer: Import adguardhome dependencies at the top

* :ambulance: Converted service handlers to be async

* :fire: Removed init step from config flow

Emilv2 committed:
Fix typo in integration component (#24250)

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

Kane610 committed:
Replace pyunifi with aiounifi in UniFi device tracker (#24149)

* Replace pyunifi with aiounifi

* Fix tests

* Add sslcontext

* Fix tests

* Fix import order

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

balloob committed:
Updated frontend to 20190602.0

balloob committed:
Add restore state to OwnTracks device tracker (#24256)

* Add restore state to OwnTracks device tracker

* Lint

* Also store entity devices

* Update test_device_tracker.py

isabellaalstrom committed:
Add service for adding event to google component (#22473)

* Add service for adding event to google component

* Add service examples for google.add_event

* add refactoring based on reviews

* Fix too long lines

* Order import

* Move to move line

* Remove parenthesis

* Add service for adding event to google component

* Add service examples for google.add_event

* add refactoring based on reviews

* Add check for correct scopes, otherwise re-authenticate

* fix build failure

* fix build failure

balloob committed:
Add restore state to Geofency (#24268)

* Add restore state to Geofency

* Lint

balloob committed:
Mobile app device tracker to restore state (#24266)

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

OttoWinter committed:
Bump aioesphomeapi to 2.1.0 (#24278)

* Bump aioesphomeapi to 2.1.0

* Update requirements txt

fabaff committed:
Remove icon() (#24280)

Kane610 committed:
deCONZ migrate to SSDP discovery (#24252)

* Migrate deCONZ to use new SSDP discovery
Add new discovery info manufacturer URL to be able to separate Hue and deCONZ bridges

* Mark deCONZ as migrated in Discovery component

* Fix tests

* Fix Hue discovery ignore deCONZ bridge

* Less snake more badger

* Mushroom

* Fix indentation

* Config flow ignore manufacturer url that is not philips

Swamp-Ig committed:
Split devices of nodes with multiple instances (#24032)

* Split devices of nodes with multiple instances

* Note entities to register with device registry

* Use EntityPlatform

balloob committed:
Add a discovery config flow to Wemo (#24208)

balloob committed:
Fix cors on the index view (#24283)

balloob committed:
Remove deps folder in config when on Docker (#24284)

* Remove deps folder in config

* Fix tests

* Fix tests with docker check

piitaya committed:
Add temperature sensor support to google smarthome thermostat device (#24264)

* Add temperature sensor support to google smarthome thermostat device

* fix lint for trait_test

* Reset temperature unit in tests

* Address comment

brandond committed:
Update pyvera to 0.3.1 for alert support (#24289)

balloob committed:
Updated frontend to 20190604.0

balloob committed:
Update translations

balloob committed:
Guard against bad states in Mobile App/OwnTracks (#24292)

balloob committed:
Run SSDP discovery in parallel (#24299)

pvizeli committed:
Create progress file for pip installs (#24297)

* Create progress file for pip installs

* fix dedlock

* unflacky test

* Address comments

* Lint

* Types

balloob committed:
Upgrade Zeroconf to 0.23 (#24300)

gibman committed:
Expose specific device_class for Velux covers (#24279)

* Update cover.py

Blinds, Rollingshutters and Awnings did not set their respective device_class attribute
Previously they would all appear as device_class "window"

* fallback device class is always 'window'

fallback device class is always 'window' in the event we have an unknown cover type

* trailing whitespace removed, trimmed as well

* Update cover.py

balloob committed:
Fix OwnTracks race condition (#24303)

* Fix OwnTracks race condition

* Lint

Kane610 committed:
address is deprecated in favor of addresses (#24302)

rolfberkenbosch committed:
Change meteoalertapi to version 0.1.3 (#24307)

* Change meteoalertapi to version 0.1.3

* Add requirements_all.txt file

pvizeli committed:
Create azure-pipelines-wheels.yml

pvizeli committed:
Update and rename azure-pipelines.yml to azure-pipelines-release.yml

pvizeli committed:
Rename azure-pipelines-release.yml to azure-pipelines.yml

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

pvizeli committed:
Rename azure-pipelines.yml to azure-pipelines-release.yml

scop committed:
Upgrade pytest and -cov (#24258)

* Upgrade pytest to 4.5.0

https://docs.pytest.org/en/latest/changelog.html#pytest-4-5-0-2019-05-11
https://docs.pytest.org/en/latest/changelog.html#pytest-4-4-2-2019-05-08

* Upgrade pytest to 4.6.0

https://docs.pytest.org/en/latest/changelog.html#pytest-4-6-0-2019-05-31

* Upgrade pytest-cov to 2.7.1

https://github.com/pytest-dev/pytest-cov/blob/master/CHANGELOG.rst#271-2019-05-03
https://github.com/pytest-dev/pytest-cov/blob/master/CHANGELOG.rst#270-2019-05-03

* Upgrade pytest to 4.6.1

https://docs.pytest.org/en/latest/changelog.html#pytest-4-6-1-2019-06-02

alengwenus committed:
Add LCN scene platform (#24242)

felipediel committed:
Add support for learning new commands (#23888)

* Add support for learning new commands

This update creates a generic service in the 'remote' component to enable remote control platforms to learn new commands.

* Update __init__.py with the proposed changes

- Add 'supported_features' property and a constant related to the 'learn_command' functionality.
- Redefine 'async_learn_command' function as a coroutine.

* Update __init__.py

* Fix assertion error

Adding the 'supported_features' attribute generated an assertion error on the 'Demo Remote' platform. This update fixes this.

* Fix duplicated 'hass' object

This update fixes a typo that occurred at the last update.

aequitas committed:
Upgrade to newer version of rflink with improve error handling on incoming data. (#24263)

Related: https://github.com/home-assistant/home-assistant/issues/23942

drobtravels committed:
SolarEdge Local Component (#23996)

* Basic local SolarEdge monitoring for energy / power

* Basic local SolarEdge monitoring for energy / power

* generated CODEOWNERS, requirements, excluded tests

* generated CODEOWNERS, requirements, excluded tests

* lint fixes, etc

* lint fixes, etc

* Fix docstyle for init

Of course thats the file I forgot to run tests on

* Load all sensors by default

They use the same API endpoint.  This changes was made per https://github.com/home-assistant/architecture/pull/244

* remve unneded date/time

* ran hassfest again

* add throttle when updating

* readd solax, mistakenly removed

* Update sensor.py

Kane610 committed:
Don't let zeroconf be smart with addresses (#24321)

scarface-4711 committed:
Push to version 0.7.9 of denonavr (#24260)

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

bachya committed:
Bump simplipy to 3.4.2 (#24326)

* Bump simplipy to 3.4.2

* Updated requirements

victorcerutti committed:
Meteofrance fix 24244 (#24315)

* Update meteofrance package version

Fix https://github.com/home-assistant/home-assistant/issues/24244

* Add code owner to manifest

* Update CODEOWNERS

jjlawren committed:
Only update media icon when necessary (#24324)

* Only update media icon when necessary

* Lint

* Comment

andre-richter committed:
components/cover: Typo in docstring. (#24329)

cgtobi committed:
Add exception handling for Netatmo climate (#24311)

* Add exception handling

* Make pylint happy

balloob committed:
Initiate websession inside event loop (#24331)

Commit:
Remove deprecated AlarmControlPanel (#24322)

presslab-us commented:
Is this waiting for something on my end?  I have no idea about the failing "codecov/patch" above, it looks like the errors are all in files that I have not touched.

dmulcahey committed:
Use node descriptor from Zigpy for ZHA (#24316)

* use zigpy node descriptor

* cleanup

dmulcahey commented:
@presslab-us trying to figure out what to do w/ this... maybe rebase the PR?

presslab-us committed:
Merge branch 'zha-implement-closure' of https://github.com/presslab-us/home-assistant into zha-implement-closure

presslab-us commented:
Looks like it's all screwed up now, sigh.  I'll just close this.

dmulcahey commented:
You can reset it and try again... ping me on discord if you want help. Same handle

presslab-us commented:
It looks fine on my repo.  I created a new pull request here: https://github.com/home-assistant/home-assistant/pull/24344.  Crossing fingers that it passes the tests.
